doc/sphinxext/display_est_repr_html.py

```diff
@@ -0,0 +1,66 @@
+"""
+Primary used to display the html output of `_repr_html_` of estimators
+"""
+import sys
+from sphinx.util.docutils import SphinxDirective
+from docutils import nodes
+from io import StringIO
+
+
+class DisplayEstimatorRepr(SphinxDirective):
+    "Execute Python code and includes stdout as HTML"
+
+    has_content = True
+    required_arguments = 0
+    optional_arguments = 0
+
+    def execute(self, code, format_str):
+        code_parts = code.split('\n')
+        final_output = code_parts[-1]
+        code_parts[-1] = format_str.format(final_output)
+        code = '\n'.join(code_parts)
+        orig_stdout, orig_stderr = sys.stdout, sys.stderr
+
+        output, err = StringIO(), StringIO()
+
+        sys.stdout, sys.stderr = output, err
+        exec(code)
+        sys.stdout, sys.stderr = orig_stdout, orig_stderr
+
+        return f"{output.getvalue()}{err.getvalue()}"
+
+    def run(self):
+        output = []
+        code = "\n".join(self.content)
+        html_repr = self.execute(code, format_str='print({}._repr_html_())')
+
+        input_code = nodes.literal_block(code, code)
```

Comment:
naming the keyword would probably make it easier for the reader so that they don't have to check the `literal_block` docstring.